src/blurb/blurb.py

```diff
@@ -868,10 +876,115 @@ def find_editor():
+    return match.group(1)
+
+
+def _extract_section_name(section):
+    if section is None:
+        return None
+
+    section = raw_section = section.strip()
+    if section.strip('+-').isdigit():
+        section_index = int(section) - 1
+        if not (0 <= section_index < len(sections)):
+            sys.exit(f"Invalid section ID: {int(section)}\n\n"
+                     f"Choose from the following table:\n\n"
+                     f'{sections_table}')
+        return sections[section_index]
+
+    if not section:
+        sys.exit(f"Empty section name!")
+
+    sanitized = re.sub(r'[_-]', ' ', section)
+    section_words = re.split(r'\s+', sanitized)
+    section_pattern = '[_ ]'.join(map(re.escape, section_words))
+    section_re = re.compile(section_pattern, re.I)
+
+    matches = []
+    for section_name in sections:
+        if section_re.match(section_name):
+            matches.append(section_name)
+
+    if not matches:
+        sys.exit(f"Invalid section name: {raw_section}\n\n"
+                 f"Choose from the following table:\n\n"
+                 f'{sections_table}')
+
+    if len(matches) > 1:
+        multiple_matches = ', '.join(map(repr, sorted(matches)))
+        sys.exit(f"More than one match for: {raw_section}\n"
+                 f"Matches: {multiple_matches}\n\n"
+                 f"Choose from the following table:\n\n"
+                 f'{sections_table}')
+
+    return matches[0]
+
+
+def _update_blurb_template(issue, section):
+    issue_number = _extract_issue_number(issue)
+    section_name = _extract_section_name(section)
+
+    # Ensure that a whitespace is given after 'gh-issue:'
+    # to help filling up the template, unless an issue
+    # number was manually specified via the CLI.
+    text = template
+
+    issue_line = ".. gh-issue:"
+    pattern = "\n" + issue_line + "\n"
+    if issue_number is None:
+        if pattern not in text:
+            sys.exit("Can't find gh-issue line to ensure there's a space on the end!")
+        replacement = "\n" + issue_line + " \n"
+    else:
+        if pattern not in text:
+            sys.exit("Can't find gh-issue line to fill!")
+        replacement = "\n" + issue_line + " " + str(issue_number) + "\n"
+
+    text = text.replace(pattern, replacement)
+
+    # Uncomment the section if needed.
+    if section_name is not None:
+        pattern = f'#.. section: {section_name}'
+        text = text.replace(pattern, pattern.lstrip('#'))
+
+    return text
+
+
 @subcommand
-def add():
+def add(*, issue=None, section=None):
     """
 Add a blurb (a Misc/NEWS.d/next entry) to the current CPython repo.
+
+Use -i/--issue to specify a GitHub issue number or link, e.g.:
+
+    blurb add -i 109198
+    # or equivalently
+    blurb add -i https://github.com/python/cpython/issues/109198
+
+The blurb's section can be specified via -s/--section
+with its ID or name (case insenstitive), e.g.:
+
+    blurb add -s %(section_example_name)r
```

Comment:
I think you can hard-code the example section name here.

Reply:
May I keep it non-hardcoded just in case we modify the name of the section for some reason in the future and forget to update the docstring?